src/utils/dates.js

```diff
@@ -11,13 +11,18 @@ export function getFutureDate(offset) {
 	return new Date(Date.now() + offset * ONE_DAY_IN_MILLISECONDS);
 }
 
+// transforms Firebase date object to JS date object
+export function transformToJSDate(date) {
+	return date?.toDate();
+}
+
 /**
  * startDate and endDate are required to be two JavaScript Date objects in order for the compare function to work
  * @param {Date} startDate
  * @param {Date} endDate
  */
 export function getDaysBetweenDates(startDate, endDate) {
-	let timeDifference = endDate.getTime() - startDate.getTime();
+	let timeDifference = endDate?.getTime() - startDate?.getTime();
 	let totalDays = Math.abs(timeDifference / ONE_DAY_IN_MILLISECONDS);
```

Comment:
will this throw an undefined/ null pointer error now with ?

Reply:
This function is meant to only accept valid JS Date Objects and passing in a null value would not produce an accurate calculation. Would a try/catch block be effective at directing an error if a null/undefined value is passed in?